File: kotlin/src/main/kotlin/com/github/lusing/qimen/LiuQin.kt
package com.github.lusing.qimen


open class LiuQin constructor(lq: Int) {
    var liuqin = XIONGDI

    init {
        this.liuqin = lq
    }

    fun getName(): String {
        when (this.liuqin) {
            FUMU -> return "父母"

            ZISUN -> return "子孙"

            GUANGUI -> return "官鬼"

            QICAI -> return "妻财"

            XIONGDI -> return "兄弟"
        }
        return ""
    }

    override fun equals(other: Any?): Boolean {
        return if (other!! is LiuQin) {
            this.liuqin == (other as LiuQin).liuqin
        } else {
            false
        }
    }

    companion object {
        const val XIONGDI = 0
        const val FUMU = 1
        const val ZISUN = 2
        const val GUANGUI = 3
        const val QICAI = 4

    }
}

File: js/liuqin.js
class LiuQin {
    constructor(lq) {
        this.liuqin = LiuQin.XIONGDI;
        this.liuqin = lq;
    }

    getName() {
        switch (this.liuqin) {
            case LiuQin.FUMU:
                return "父母";
            case LiuQin.ZISUN:
                return "子孙";
            case LiuQin.GUANGUI:
                return "官鬼";
            case LiuQin.QICAI:
                return "妻财";
            case LiuQin.XIONGDI:
                return "兄弟";
            default:
                return "";
        }
    }

    equals(other) {
        if (other instanceof LiuQin) {
            return this.liuqin === other.liuqin;
        } else {
            return false;
        }
    }

    static get XIONGDI() {
        return 0;
    }

    static get FUMU() {
        return 1;
    }

    static get ZISUN() {
        return 2;
    }

    static get GUANGUI() {
        return 3;
    }

    static get QICAI() {
        return 4;
    }
}

module.exports = LiuQin;